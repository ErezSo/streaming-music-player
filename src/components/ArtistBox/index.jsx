import React from "react";
import { Link } from "react-router-dom";

const styles = {
  artistLink: {
    textDecoration: "none"
  },
  groupItem: {
    width: "100%",
    textAlign: "left",
    cursor: "pointer"
  },
  artistImage: {
    height: "80px",
    width: "80px"
  },
  artistDetails: {
    display: "inline-block"
  },
  artistName: {
    fontSize: "1.5em",
    paddingLeft: "1em"
  },
  countWrapper: {
    listStyleType: "none",
    margin: 0,
    padding: 0,
    float: "left",
    marginLeft: "22px"
  },
  playCount: {
    fontSize: "11px"
  }
};

const ArtistBox = ({ artist }) => {
  return (
    <Link to={{ pathname: `/songs/${artist.id}` }} style={styles.artistLink}>
      <button
        type="button"
        className="list-group-item"
        style={styles.groupItem}
      >
        <img
          src={artist.thumb}
          alt="artist"
          className="img-thumbnail"
          style={styles.artistImage}
        />
        <div style={styles.artistDetails}>
          <div style={styles.artistName}>{artist.user.username}</div>
          <ul style={styles.countWrapper}>
            <li style={styles.playCount}>
              Play Count: {artist.playback_count}
            </li>
          </ul>
        </div>
      </button>
    </Link>
  );
};

export default ArtistBox;
