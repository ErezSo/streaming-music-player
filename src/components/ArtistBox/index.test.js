import React from "react";
import { shallow } from "enzyme";
import ArtistBox from "./";

describe("ArtistBox", () => {
  let wrapper;
  const artist = {
    id: "43",
    title: "Bestami Turna - Soundpark 14.04.2018",
    thumb:
      "https://images.hearthis.at/1/5/2/_/uploads/283048/image_track/1914537/w200_h200_q70_----1525319281387.jpg",
    user: {
      username: "TDSmix"
    },
    playback_count: "126"
  };

  beforeEach(() => {
    wrapper = shallow(<ArtistBox artist={artist} />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
