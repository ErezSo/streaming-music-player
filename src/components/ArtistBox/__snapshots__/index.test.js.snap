// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistBox renders correctly 1`] = `
"<Link to={{...}} style={{...}} replace={false}>
  <button type=\\"button\\" className=\\"list-group-item\\" style={{...}}>
    <img src=\\"https://images.hearthis.at/1/5/2/_/uploads/283048/image_track/1914537/w200_h200_q70_----1525319281387.jpg\\" alt=\\"artist\\" className=\\"img-thumbnail\\" style={{...}} />
    <div style={{...}}>
      <div style={{...}}>
        TDSmix
      </div>
      <ul style={{...}}>
        <li style={{...}}>
          Play Count: 
          126
        </li>
      </ul>
    </div>
  </button>
</Link>"
`;
